### PR TITLE
Fix pull_image operation & add basic test

### PR DESCRIPTION
**Pull request by oritstone**
Signed-off-by: Orit Stone <orit.stone@hp.com>

### CODE REVIEW

File: content/io/cloudslang/docker/images/pull_image.sl [hunk 1]
```diff
@@ -36,24 +36,27 @@ operation:
     - imageName
     - host
     - port:
-        default: "'22'"
+        required: false
     - username
     - password
```
Comment (meirwah) [inline content/io/cloudslang/docker/images/pull_image.sl]: maybe password too is required: false, since you either need privateKeyFile or password

**Comment by meirwah** [comment]
default values were given for documentation proposes  (just remembered ) as always , good reason for both ways here.. no clear preference

File: test/io/cloudslang/docker/images/test_pull_image.sl [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            - username
+            - password
+            - imageName: image_name
+        publish:
+          - return_result
+          - error_message
+        navigate:
+          SUCCESS: clear_image
+          FAILURE: FAIL_PULL_IMAGE
+
```
Comment (meirwah) [inline test/io/cloudslang/docker/images/test_pull_image.sl]: add here step get_all_images to verify you really got the image downloaded

File: test/io/cloudslang/docker/images/test_pull_image.sl [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    - validate_ssh:
+        do:
+          linux.validate_linux_machine_ssh_access:
+            - host
+            - port
+            - username
+            - password
+        navigate:
+          SUCCESS: pull_image
+          FAILURE: FAIL_VALIDATE_SSH
```
Comment (meirwah) [inline test/io/cloudslang/docker/images/test_pull_image.sl]: And add a step before (get_all_images).. to verify the image "image_name" is NOT present

File: test/io/cloudslang/docker/images/pull_image.inputs.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+#
+#   The Apache License is available at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+####################################################
+testDocker_pull_image_Success:
+  inputs:
+    - host: localhost
+    - port: "49153"
+    - username: root
```
Comment (orius123) [inline test/io/cloudslang/docker/images/pull_image.inputs.yaml]: we need to extract this properties somehow

File: test/io/cloudslang/docker/images/test_pull_image.sl [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    - host
+    - port:
+        required: false
+    - username
+    - password
+    - image_name
+
+  workflow:
+
+    - validate_ssh:
```
Comment (orius123) [inline test/io/cloudslang/docker/images/test_pull_image.sl]: this step should be part of the ssh op

File: test/io/cloudslang/docker/images/test_pull_image.sl [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  inputs:
+    - host
+    - port:
+        required: false
+    - username
+    - password
+    - image_name
+
+  workflow:
+
```
Comment (orius123) [inline test/io/cloudslang/docker/images/test_pull_image.sl]: we should add some set-up subflow here

File: test/io/cloudslang/docker/images/test_pull_image.sl [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+            - port
+            - username
+            - password
+        publish:
+          - image_list
+        navigate:
+          SUCCESS: verify_no_images_before
+          FAILURE: FAIL_GET_ALL_IMAGES_BEFORE
+
+    - verify_no_images_before:
```
Comment (orius123) [inline test/io/cloudslang/docker/images/test_pull_image.sl]: this 2 'before' tasks should be moved to the set-up subflow

File: test/io/cloudslang/docker/images/test_pull_image.sl [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+          FAILURE: MACHINE_IS_NOT_CLEAN
+
+    - pull_image:
+        do:
+          images.pull_image:
+            - host
+            - port
+            - username
+            - password
+            - imageName: image_name
```
Comment (orius123) [inline test/io/cloudslang/docker/images/test_pull_image.sl]: why is this camelcase?

File: test/io/cloudslang/docker/images/test_pull_image.sl [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+          - image_list
+        navigate:
+          SUCCESS: verify_image_name
+          FAILURE: FAIL_GET_ALL_IMAGES
+
+    - verify_image_name:
+        do:
+          strings.string_occurrence_counter:
+            - string_in_which_to_search: image_list
+            - string_to_find: image_name + ":latest"
```
Comment (orius123) [inline test/io/cloudslang/docker/images/test_pull_image.sl]: can't you use the outputs check of the diphthong framework for that?

File: test/io/cloudslang/docker/images/test_pull_image.sl [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+            - port
+            - username
+            - password
+            - images: image_name
+        navigate:
+          SUCCESS: SUCCESS
+          FAILURE: FAIL_CLEAR_IMAGE
+
+  outputs:
+    - return_result
```
Comment (orius123) [inline test/io/cloudslang/docker/images/test_pull_image.sl]: why are you outputing it?

File: test/io/cloudslang/docker/images/pull_image.inputs.yaml (hunk 4 above)
Comment (meirwah) [inline test/io/cloudslang/docker/images/pull_image.inputs.yaml]: flow inputs file?

**Comment by meirwah** [comment]
:+1:

**Pull request merged**